### ISSUE TITLE
Bogus report about subprocess.Popen.stdin

### ISSUE THREAD

**Issue by keszybz** [python/mypy#10583]
This error started being reported recently (mypy-0.812 didn't have it, 0.820+dev.b562cc235dcd2e71d2fed1aad72a4a8400bc307d does):
```console
$ cat repro.py
import subprocess
with subprocess.Popen(['cat'], stdin=subprocess.PIPE, stdout=subprocess.DEVNULL) as cat:
    cat.stdin.write(b'asdf')
    cat.stdin.close()
$ mypy repro.py
repro.py:3:5: error: Item "None" of "Optional[IO[bytes]]" has no attribute "write"  [union-attr]
        cat.stdin.write(b'asdf')
        ^
repro.py:4:5: error: Item "None" of "Optional[IO[bytes]]" has no attribute "close"  [union-attr]
        cat.stdin.close()
        ^
Found 2 errors in 1 file (checked 1 source file)
```
`stdin` is of type `subprocess.PIPE`, so even if it were closed, the object still has `.write()` and `.close()` methods, they may just throw an error if called.

**Comment by srittau** [comment]
Introduced in python/typeshed#3652. No good way around it, except by making `Popen` generic over the std streams. Maybe we should just do that?

**Comment by ndevenish** [comment]
Same issue with `for line in process.stdout:`, if requesting that in the `Popen` constructor. Are we really supposed to mangle all of our code to satisfy the type checker here?

**Comment by hauntsaninja** [comment]
Closing, since typeshed changed this years ago and I haven't heard too many complaints. typeshed is the better repo to discuss this anyway
